kani-compiler/src/kani_middle/transform/check_aliasing/instrumentation.rs

```diff
@@ -0,0 +1,446 @@
+                body.insert_terminator(source, InsertPosition::After, terminator.clone());
+                let execute_ghost_block = source.clone();
+
+                // Instrument enter ghost:
+                let span = original_span;
+                let target = execute_ghost_block.bb();
+                let kind = TerminatorKind::Goto { target };
+                let terminator = Terminator { kind, span };
+                body.replace_terminator(&enter_ghost_block, terminator);
+                // Instrument execute ghost:
+                let target = execute_terminator_block.bb();
+                let kind = TerminatorKind::Goto { target };
+                let terminator = Terminator { kind, span };
+                body.replace_terminator(&execute_ghost_block, terminator);
+                // Instrument execute terminator
+                body.replace_terminator(&execute_terminator_block, original);
+
+                self.ghost_index = execute_ghost_block;
+            }
+            SourceInstruction::Statement { idx, bb } => {
+                // In this case it is simple, merely goto the ghost code
+                // immdediately.
+                let span = self.body.blocks()[bb].statements[idx].span;
+                let target = self.body.blocks().len();
+                let kind = TerminatorKind::Goto { target };
+                let terminator = Terminator { kind, span };
+                let min_processed = &mut self.min_processed.clone();
+                self.body.insert_terminator(min_processed, InsertPosition::After, terminator);
+                self.ghost_index = *min_processed;
+            }
+        }
+    }
+
+    /// Instrument a stack reference of the fo
+    /// lvalue = &rvalue
+    /// with an update to the stacked borrows state,
+    /// at the code index source.
+    pub fn instrument_new_stack_reference(&mut self, lvalue: Local, rvalue: Local) -> Result<()> {
+        // Initialize the constants
+        let ty = self.body.locals()[rvalue].ty;
+        let lvalue_ref = self.meta_stack.get(&lvalue).unwrap();
+        let rvalue_ref = self.meta_stack.get(&rvalue).unwrap();
+        self.call(
+            Signature::new("KaniNewMutRefFromValue", &[GenericArgKind::Type(ty)]),
+            vec![*lvalue_ref, *rvalue_ref],
+            self.unit,
+        )?;
+        Ok(())
+    }
+
+    /// Instrument with stack violated / not violated
+    pub fn instrument_stack_check(&mut self) -> Result<()> {
+        let span = match self.min_processed {
+            SourceInstruction::Statement { idx, bb } => self.body.blocks()[bb].statements[idx].span,
+            SourceInstruction::Terminator { bb } => self.body.blocks()[bb].terminator.span,
+        };
+        self.call(Signature::new("KaniStackValid", &[]), vec![], self.valid)?;
+        let msg = format!(
+            "Stacked borrows aliasing model violated at {:?}:{:?}",
+            span.get_filename(),
+            span.get_lines()
```

Comment:
I might be wrong, but doesn't the failed assertion contain the location by default?